core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -146,64 +147,86 @@ class BrokerTopicMetrics(name: Option[String]) extends KafkaMetricsGroup {
     case Some(topic) => Map("topic" -> topic)
   }
 
+  case class MeterWrapper(metricType: String, eventType: String) {
+    @volatile private var lazyMeter: Meter = _
+    private val meterLock = new Object
+
+    def meter(): Meter = {
+      var meter = lazyMeter
+      if (meter == null) {
+        meterLock synchronized {
+          if (lazyMeter == null)
+            lazyMeter = newMeter(metricType, eventType, TimeUnit.SECONDS, tags)
+          meter = lazyMeter
```

Comment:
Nit: the standard practice is to assign the local variable first and then assign the field from the local variable. It's the same code and a tiny bit more efficient.